huaweicloud/services/rds/resource_huaweicloud_rds_cross_region_backup_strategy.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	getBackupStrategyPath := getBackupStrategyClient.Endpoint + getBackupStrategyHttpUrl
+	getBackupStrategyPath = strings.ReplaceAll(getBackupStrategyPath, "{project_id}",
+		getBackupStrategyClient.ProjectID)
+	getBackupStrategyPath = strings.ReplaceAll(getBackupStrategyPath, "{instance_id}", d.Id())
+
+	getBackupStrategyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	getBackupStrategyResp, err := getBackupStrategyClient.Request("GET", getBackupStrategyPath, &getBackupStrategyOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS cross region backup strategy")
+	}
+
+	getBackupStrategyRespBody, err := utils.FlattenResponse(getBackupStrategyResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	policyPara := utils.PathSearch("policy_para", getBackupStrategyRespBody, nil)
+	if policyPara == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	backupStrategies := policyPara.([]interface{})
+	if len(backupStrategies) == 0 || utils.PathSearch("keep_days", backupStrategies[0], 0).(float64) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", d.Id()),
+		d.Set("backup_type", utils.PathSearch("backup_type", backupStrategies[0], nil)),
+		d.Set("keep_days", utils.PathSearch("keep_days", backupStrategies[0], nil)),
+		d.Set("destination_region", utils.PathSearch("destination_region", backupStrategies[0], nil)),
+		d.Set("destination_project_id", utils.PathSearch("destination_project_id", backupStrategies[0], nil)),
+	)
+
+	if len(backupStrategies) == 1 {
+		mErr = multierror.Append(mErr, d.Set("backup_type", "auto"))
+	} else {
+		mErr = multierror.Append(mErr, d.Set("backup_type", "all"))
+	}
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceBackupStrategyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateBackupStrategyChanges := []string{
+		"backup_type",
+		"keep_days",
+	}
+
+	if d.HasChanges(updateBackupStrategyChanges...) {
+		// updateBackupStrategy: update a RDS cross region backup strategy.
+		var (
+			updateBackupStrategyProduct = "rds"
+		)
+		updateBackupStrategyClient, err := cfg.NewServiceClient(updateBackupStrategyProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating RDS client: %s", err)
+		}
+
+		if d.HasChange("keep_days") {
+			err = updateCrossRegionBackupStrategy(ctx, updateBackupStrategyClient, d.Timeout(schema.TimeoutUpdate),
+				d.Get("instance_id").(string), buildUpdateBackupStrategyKeepDaysBodyParams(d))
+			if err != nil {
+				return diag.Errorf("error updating RDS cross region backup strategy: %s", err)
+			}
+		}
+		// if backup_type is changed and the value is auto, it indicates the original value is all, we should close the
+		// incremental backup
```

Comment:
```suggestion
		// If backup_type is changed from 'all' to 'auto', we should close the incremental backup.
```
maybe it is easier to understand

Reply:
fixed